From b2562db7d0657e48baa22352acc9102c3f1a0feb Mon Sep 17 00:00:00 2001
From: Tomas Pecka <peckato1@users.noreply.github.com>
Date: Tue, 13 Jul 2021 16:37:08 +0200
Subject: [PATCH] Move sd_lldp to libsystemd

This makes sd_lldp public in libsystemd. Clients can therefore operate
with LLDP structures and files (e.g. parse LLDP neighbours).

Based on libsystemd v247.3
---
 meson.build                                   |  1 +
 src/libsystemd-network/meson.build            |  6 ---
 src/libsystemd/libsystemd.sym                 | 39 +++++++++++++++++++
 src/libsystemd/meson.build                    |  6 +++
 .../sd-lldp}/lldp-internal.h                  |  0
 .../sd-lldp}/lldp-neighbor.c                  |  0
 .../sd-lldp}/lldp-neighbor.h                  |  0
 .../sd-lldp}/lldp-network.c                   |  0
 .../sd-lldp}/lldp-network.h                   |  0
 .../sd-lldp}/sd-lldp.c                        |  0
 .../sd-lldp}/test-lldp.c                      |  0
 src/systemd/meson.build                       |  2 +-
 src/test/meson.build                          |  9 ++---
 13 files changed, 51 insertions(+), 12 deletions(-)
 rename src/{libsystemd-network => libsystemd/sd-lldp}/lldp-internal.h (100%)
 rename src/{libsystemd-network => libsystemd/sd-lldp}/lldp-neighbor.c (100%)
 rename src/{libsystemd-network => libsystemd/sd-lldp}/lldp-neighbor.h (100%)
 rename src/{libsystemd-network => libsystemd/sd-lldp}/lldp-network.c (100%)
 rename src/{libsystemd-network => libsystemd/sd-lldp}/lldp-network.h (100%)
 rename src/{libsystemd-network => libsystemd/sd-lldp}/sd-lldp.c (100%)
 rename src/{libsystemd-network => libsystemd/sd-lldp}/test-lldp.c (100%)

diff --git a/meson.build b/meson.build
index 580964c3fa..22783a2f9f 100644
--- a/meson.build
+++ b/meson.build
@@ -1574,6 +1574,7 @@ includes = include_directories('src/basic',
                                'src/libsystemd/sd-event',
                                'src/libsystemd/sd-hwdb',
                                'src/libsystemd/sd-id128',
+                               'src/libsystemd/sd-lldp',
                                'src/libsystemd/sd-netlink',
                                'src/libsystemd/sd-network',
                                'src/libsystemd/sd-resolve',
diff --git a/src/libsystemd-network/meson.build b/src/libsystemd-network/meson.build
index 604cfd999b..7864fc3d07 100644
--- a/src/libsystemd-network/meson.build
+++ b/src/libsystemd-network/meson.build
@@ -34,12 +34,6 @@ sources = files('''
         sd-dhcp6-lease.c
         dhcp-identifier.h
         dhcp-identifier.c
-        lldp-internal.h
-        lldp-network.h
-        lldp-network.c
-        lldp-neighbor.h
-        lldp-neighbor.c
-        sd-lldp.c
 '''.split())

 network_internal_h = files('network-internal.h')
diff --git a/src/libsystemd/libsystemd.sym b/src/libsystemd/libsystemd.sym
index f83b364c96..45565e5821 100644
--- a/src/libsystemd/libsystemd.sym
+++ b/src/libsystemd/libsystemd.sym
@@ -735,4 +735,43 @@ global:
         sd_device_get_current_tag_next;
         sd_device_has_current_tag;
         sd_device_set_sysattr_valuef;
+
+        sd_lldp_start;
+        sd_lldp_neighbor_ref;
+        sd_lldp_neighbor_unref;
+        sd_lldp_neighbor_get_source_address;
+        sd_lldp_neighbor_get_destination_address;
+        sd_lldp_neighbor_get_raw;
+        sd_lldp_neighbor_get_chassis_id;
+        sd_lldp_neighbor_get_chassis_id_as_string;
+        sd_lldp_neighbor_get_port_id;
+        sd_lldp_neighbor_get_port_id_as_string;
+        sd_lldp_neighbor_get_ttl;
+        sd_lldp_neighbor_get_system_name;
+        sd_lldp_neighbor_get_system_description;
+        sd_lldp_neighbor_get_port_description;
+        sd_lldp_neighbor_get_mud_url;
+        sd_lldp_neighbor_get_system_capabilities;
+        sd_lldp_neighbor_get_enabled_capabilities;
+        sd_lldp_neighbor_from_raw;
+        sd_lldp_neighbor_tlv_rewind;
+        sd_lldp_neighbor_tlv_next;
+        sd_lldp_neighbor_tlv_get_type;
+        sd_lldp_neighbor_tlv_is_type;
+        sd_lldp_neighbor_tlv_get_oui;
+        sd_lldp_neighbor_tlv_is_oui;
+        sd_lldp_neighbor_tlv_get_raw;
+        sd_lldp_neighbor_get_timestamp;
+        sd_lldp_start;
+        sd_lldp_stop;
+        sd_lldp_attach_event;
+        sd_lldp_detach_event;
+        sd_lldp_get_event;
+        sd_lldp_set_callback;
+        sd_lldp_set_ifindex;
+        sd_lldp_new;
+        sd_lldp_get_neighbors;
+        sd_lldp_set_neighbors_max;
+        sd_lldp_match_capabilities;
+        sd_lldp_set_filter_address;
 } LIBSYSTEMD_246;
diff --git a/src/libsystemd/meson.build b/src/libsystemd/meson.build
index 50716f7b94..68b19d30c5 100644
--- a/src/libsystemd/meson.build
+++ b/src/libsystemd/meson.build
@@ -70,6 +70,12 @@ libsystemd_sources = files('''
         sd-hwdb/hwdb-util.c
         sd-hwdb/hwdb-util.h
         sd-hwdb/sd-hwdb.c
+        sd-lldp/lldp-internal.h
+        sd-lldp/lldp-network.h
+        sd-lldp/lldp-network.c
+        sd-lldp/lldp-neighbor.h
+        sd-lldp/lldp-neighbor.c
+        sd-lldp/sd-lldp.c
         sd-netlink/generic-netlink.c
         sd-netlink/generic-netlink.h
         sd-netlink/netlink-internal.h
diff --git a/src/libsystemd-network/lldp-internal.h b/src/libsystemd/sd-lldp/lldp-internal.h
similarity index 100%
rename from src/libsystemd-network/lldp-internal.h
rename to src/libsystemd/sd-lldp/lldp-internal.h
diff --git a/src/libsystemd-network/lldp-neighbor.c b/src/libsystemd/sd-lldp/lldp-neighbor.c
similarity index 100%
rename from src/libsystemd-network/lldp-neighbor.c
rename to src/libsystemd/sd-lldp/lldp-neighbor.c
diff --git a/src/libsystemd-network/lldp-neighbor.h b/src/libsystemd/sd-lldp/lldp-neighbor.h
similarity index 100%
rename from src/libsystemd-network/lldp-neighbor.h
rename to src/libsystemd/sd-lldp/lldp-neighbor.h
diff --git a/src/libsystemd-network/lldp-network.c b/src/libsystemd/sd-lldp/lldp-network.c
similarity index 100%
rename from src/libsystemd-network/lldp-network.c
rename to src/libsystemd/sd-lldp/lldp-network.c
diff --git a/src/libsystemd-network/lldp-network.h b/src/libsystemd/sd-lldp/lldp-network.h
similarity index 100%
rename from src/libsystemd-network/lldp-network.h
rename to src/libsystemd/sd-lldp/lldp-network.h
diff --git a/src/libsystemd-network/sd-lldp.c b/src/libsystemd/sd-lldp/sd-lldp.c
similarity index 100%
rename from src/libsystemd-network/sd-lldp.c
rename to src/libsystemd/sd-lldp/sd-lldp.c
diff --git a/src/libsystemd-network/test-lldp.c b/src/libsystemd/sd-lldp/test-lldp.c
similarity index 100%
rename from src/libsystemd-network/test-lldp.c
rename to src/libsystemd/sd-lldp/test-lldp.c
diff --git a/src/systemd/meson.build b/src/systemd/meson.build
index 3d328e5fd3..2e53004452 100644
--- a/src/systemd/meson.build
+++ b/src/systemd/meson.build
@@ -9,6 +9,7 @@ _systemd_headers = '''
         sd-event.h
         sd-hwdb.h
         sd-id128.h
+        sd-lldp.h
         sd-journal.h
         sd-login.h
         sd-messages.h
@@ -28,7 +29,6 @@ _not_installed_headers = '''
         sd-dhcp-server.h
         sd-ipv4acd.h
         sd-ipv4ll.h
-        sd-lldp.h
         sd-ndisc.h
         sd-netlink.h
         sd-network.h
diff --git a/src/test/meson.build b/src/test/meson.build
index 6234294947..f2fd2362da 100644
--- a/src/test/meson.build
+++ b/src/test/meson.build
@@ -1069,6 +1069,10 @@ tests += [
          [],
          []],

+        [['src/libsystemd/sd-lldp/test-lldp.c'],
+         [libsystemd_static,
+          libshared],
+         []],
 ]

 if cxx_cmd != ''
@@ -1154,11 +1158,6 @@ tests += [
          [libshared,
           libsystemd_network],
          []],
-
-        [['src/libsystemd-network/test-lldp.c'],
-         [libshared,
-          libsystemd_network],
-         []],
 ]

 ############################################################
--
2.32.0

